app/src/main/proto/vss.proto

```diff
@@ -0,0 +1,153 @@
+  //
+  //  Requests with conflicting version will fail with `ConflictException` as ErrorCode.
+  //
+  //  Considerations for transactions:
+  //  Transaction writes of multiple items have a performance overhead, hence it is recommended to use
+  //  them only if required by the client application to ensure logic/code correctness.
+  //  That is, transactionItems are not a substitute for batch-write of multiple unrelated items.
+  //  When a write of multiple unrelated items is desired, it is recommended to use separate
+  //  PutObjectRequests.
+  //
+  //  Consistency Guarantee:
+  //  All PutObjectRequests are strongly consistent i.e. they provide read-after-write and
+  //  read-after-update consistency guarantees.
+  repeated KeyValue transactionItems = 3;
+}
+
+message PutObjectResponse {
+}
+
+//  When HttpStatusCode is not ok (200), the response `content` contains a serialized ErrorResponse
+//  with the relevant ErrorCode and ErrorCode's integer value as HttpStatusCode.
+message ErrorResponse {
+
+  //  The error code uniquely identifies an error condition.
+  //  It is meant to be read and understood programmatically by code that detects/handles errors by
+  //  type.
+  ErrorCode errorCode = 1;
+
+  //  The error message contains a generic description of the error condition in English.
+  //  It is intended for a human audience only and should not be parsed to extract any information
+  //  programmatically. Client-side code may use it for logging only.
+  string message = 2;
+}
+
+//  ErrorCodes to be used in ErrorResponse along with corresponding integer values that will be used
+//  as HttpStatusCode in HttpResponse.
+enum ErrorCode {
+
+  //  Default protobuf Enum value. Will not be used as ErrorCode by server.
+  Unknown = 0;
+
+  //  ConflictException is used when the request contains mismatched version (either key or global)
+  //  in PutObjectRequest. For more info refer PutObjectRequest.
+  ConflictException = 409;
+
+  //  InvalidRequestException is used in the following cases:
+  //    - The request was missing a required argument.
+  //    - The specified argument was invalid, incomplete or in the wrong format.
+  //    - The request body of api cannot be deserialized into corresponding protobuf object.
+  InvalidRequestException = 400;
+
+  //  An internal server error occurred, client is probably at no fault and can safely retry this
+  //  error with exponential backoff.
+  InternalServerException = 500;
+}
+
+message KeyValue {
+
+  //  Key against which the value is stored.
+  string key = 1;
+
+  //  For first write of key, version should be '0'. If the write succeeds, clients must increment
+  //  their corresponding key version (client-side) by 1.
+  //  The server increments key version (server-side) for every successful write, hence this
+  //  client-side increment is required to ensure matching versions. These updated key versions should
+  //  be used in subsequent PutObjectRequests for the keys.
```

Comment:
First sentence should document what the version is rather than jumping into how to set it.